File: leetcode/C++/candy.cpp
class Solution {
public:
    int candy(vector<int>& ratings) {
        int sol=0;
        vector<int> v(ratings.size(),1);
        for(int i=1; i<ratings.size(); ++i){
            if(ratings[i-1]<ratings[i]) v[i]=v[i-1]+1;
        }
        sol=v.back();
        for(int i=(int)ratings.size()-2; i>=0; --i) {
            if(ratings[i]>ratings[i+1]) if (v[i]<v[i+1]+1) v[i]=v[i+1]+1;
            sol+=v[i];
        }
        return sol;
    }
};

File: leetcode/Python/candy.py
class Solution:
    def candy(self, ratings: List[int]) -> int:
        v = [1]*len(ratings)
        for i in range(len(ratings)-1):
            if ratings[i]<ratings[i+1]:
                v[i+1]=v[i]+1
        for i in range(len(ratings)-2,-1,-1):
            if ratings[i]>ratings[i+1]:
                v[i]=max(v[i+1]+1,v[i])
        return sum(v)